src/thor/astar.cc

```diff
@@ -434,8 +437,11 @@ void AStarPathAlgorithm::SetOrigin(GraphReader& graphreader,
     }
 
     // Store the closest node info
-    if (closest_ni == nullptr) {
+    auto nodeinfo_point= nodeinfo->latlng(tile->BoundingBox().minpt());
+    float node_distance = origin_ll.DistanceSquared(nodeinfo_point) < distance_to_closest_ni;
```

Comment:
in the end this is only measuring the distance to the closest end node, if we wanted to be really pedantic about most appropriate timezone we should probably also consider the begin nodes of the edges. of course for extreme accuracy we'd have to load the timezone sqlite db and do point in polygon, but yeah lets not do that :D

Reply:
I'm happy to scrap this more complicated logic I added and instead update the comment `// Store the closest node info` to say something along _grab the first nodeinfo for timezone estimation, exactly closest node is not important_.

How's that?

Reply:
sounds good to me. i just wanted to point the complexity out, in case you werent aware